rust/src/lockfile.rs

```diff
@@ -225,8 +226,9 @@ pub(crate) fn ror_lockfile_write(
         }),
     };
 
-    for pkg in packages {
-        let pkg_ref = unsafe { &mut *(pkg as *mut libdnf_sys::DnfPackage) };
+    for i in 0..(packages.as_mut().length()) {
+        let pkg = packages.as_mut().get(i);
+        let pkg_ref = unsafe { &mut *(&mut pkg.0 as *mut _ as *mut libdnf_sys::DnfPackage)};
```

Comment:
This bit is obviously awful (there may be a better way) - I think what would help here is to actually use the same gtk-rs [glib_wrapper!](http://gtk-rs.org/docs/glib/macro.glib_wrapper.html) macro for the GObjects, then we could use the [Cast APIs](https://docs.rs/glib/0.10.3/glib/object/trait.Cast.html) and stay away from unsafe pointer bits here.

I only realized that after I'd mostly finished this PR though and wanted to push what works now without redoing it again.

Reply:
Or to say it another way - let's try to use gtk-rs for the GObject stuff, and use cxx-rs "natively" for the native C++ stuff and use our intersection of the two - instead of hand-rolling bindings for GObject stuff again.